### PR TITLE
expression:implement function password.

### PR DESCRIPTION
**Pull request by louishust**

### CODE REVIEW

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -395,7 +395,39 @@ type builtinPasswordSig struct {
 // eval evals a builtinPasswordSig.
 // See https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_password
 func (b *builtinPasswordSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("PASSWORD")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
```
Comment (shenli) [inline expression/builtin_encryption.go]: return d, errors.Trace(err)

File: expression/builtin_encryption.go [hunk 2]
```diff
@@ -395,7 +395,39 @@ type builtinPasswordSig struct {
...
+		d.SetString("")
+		return d, nil
+	}
+
+	pass, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	if pass == "" {
```
Comment (shenli) [inline expression/builtin_encryption.go]: if length(pass) == 0

File: expression/builtin_encryption.go [hunk 3]
```diff
@@ -395,7 +395,39 @@ type builtinPasswordSig struct {
...
+
+	// Two stage SHA1 hash of the password
+	h1 := sha1.New()
+	h1.Write([]byte(pass))
+	hash1 := h1.Sum(nil)
+
+	h2 := sha1.New()
+	h2.Write(hash1)
+	hash2 := h2.Sum(nil)
+	ret := "*" + fmt.Sprintf("%X", hash2)
```
Comment (shenli) [inline expression/builtin_encryption.go]: ret := fmt.Sprintf("*%X", hash2)

File: expression/builtin_encryption.go (hunk 1 above)
Comment (louishust) [inline expression/builtin_encryption.go]: SELECT PASSWORD(NULL) return empty string instead of NULL.

File: expression/builtin_encryption.go (hunk 3 above)
Comment (louishust) [inline expression/builtin_encryption.go]: need the "*", you can test on MySQL

File: expression/builtin_encryption.go (hunk 2 above)
Comment (louishust) [inline expression/builtin_encryption.go]: I'll fix it

File: expression/builtin_encryption.go (hunk 3 above)
Comment (louishust) [inline expression/builtin_encryption.go]: I got it, make "*" in the format string

File: expression/builtin_encryption.go (hunk 1 above)
Comment (louishust) [inline expression/builtin_encryption.go]: Got it, i will fix

**Comment by louishust** [comment]
@shenli PTAL

**Comment by louishust** [comment]
move to new PR #3275

**Pull request closed without merging**